src/chrome/Button.tsx

```diff
@@ -10,59 +12,130 @@ export type ButtonType = 'primary' // light blue button with white text
+  // if icon is present, the button will be slightly taller than normal and will
+  // have an icon to the left of the children
+  icon?: string
+  // there are three sizes of button, default is 'md'
   size?: 'sm' | 'md' | 'lg'
-  className?: string
+  // submit makes it a form submit button
   submit?: boolean
+  // disabled will make the button "grayed out", no pointer on hover, onClick will do nothing
   disabled?: boolean
-  id?: string
+  // block will make the button full width and with slightly larger border radii
+  block?: boolean
+  // caret will add a small caretRight icon to the right of the children
+  caret?: boolean
+  // onClick is a function that will fire when the button is clicked
+  onClick?: () => void
 }
 
 const Button: React.FC<ButtonProps> = ({
+  id,
+  className,
   type='primary',
+  icon,
   size='md',
-  className,
-  onClick= () => {},
   submit = false,
   disabled = false,
+  block = false,
+  caret = false,
+  onClick= () => {},
   children,
-  id = undefined
-}) => (
-  <button
-    id={id}
-    type={submit ? 'submit' : 'button'}
-    style={{
-      fontSize: 11
-    }}
-    className={classNames(
-      'inline-flex items-center justify-center rounded-md shadow-sm bg-transparent font-semibold focus:outline-none focus:ring-none mt-0 transition-all duration-100',
-      className,
-      {
-        'cursor-default bg-opacity-40 hover:bg-opacity-40': disabled
-      },
-      {
-        'text-sm px-2.5 h-9 ': (size === 'sm'),
-        'text-sm px-2.5 h-9': (size === 'md'),
-        'text-md px-8 py-3': (size === 'lg'),
-      },
-      {
-        'text-white bg-qritile hover:bg-qritile-600': (type === 'primary'),
-        'text-qritile hover:text-qritile-600 text-sm font-medium border-2 border-qritile hover:border-qritile-600 box-border': (type === 'primary-outline'),
-        'text-white bg-qripink-600 hover:bg-qripink-': (type === 'secondary'),
-        'text-qripink-600 hover:text-qripink-700 text-sm font-medium border-2 border-qripink-600 hover:border-qripink-700 box-border': (type === 'secondary-outline'),
-        'text-qrigray-900 bg-warningyellow hover:bg-warningyellow-600': (type === 'warning'),
-        'text-white bg-dangerred hover:bg-dangerred-600': (type === 'danger'),
-        'text-qrigray-400 hover:text-qrigray-600 border border-qrigray-400 hover:border-qrigray-600': (type === 'light'),
-        'text-qrigray-900 hover:text-qripink border border-qrigray-900 hover:border-qripink': (type === 'dark'),
-      }
-    )}
-    onClick={onClick}
-    disabled={disabled}
-  >
-    {children}
-  </button>
-)
+}) => {
+  let fontSize = 10
+  let height = 30
+  let sizeClassName = 'rounded px-2.5'
+  let caretSize: IconSize = '3xs'
+
+  if (size === 'md') {
+    fontSize = 12
+    height = 32
+    sizeClassName = 'rounded-md px-2.5'
+    caretSize = '2xs'
+  }
+
+  if (size === 'lg') {
+    fontSize = 14
+    height = 38
+    sizeClassName = 'rounded-lg px-8'
+    caretSize = 'xs'
+  }
```

Comment:
group these with an `else if`. default the above to `md` settings (which is the default size prop), and flip the `if` conditional to check if `size === 'sm'`. This'll make fewer assignments in the default case